dev/devicelab/lib/framework/adb.dart

```diff
@@ -551,29 +551,49 @@ class IosDeviceDiscovery implements DeviceDiscovery {
     print('Device chosen: $_workingDevice');
   }
 
-  // Returns a colon-separated environment variable that contains the paths
-  // of linked libraries for idevice_id
-  Map<String, String> get _ideviceIdEnvironment {
-    final String libPath = const <String>[
-      'libimobiledevice',
-      'usbmuxd',
-      'libplist',
-      'openssl',
-      'ios-deploy',
-    ].map((String packageName) => path.join(getArtifactPath(), packageName)).join(':');
-    return <String, String>{'DYLD_LIBRARY_PATH': libPath};
-  }
-
   @override
   Future<List<String>> discoverDevices() async {
-    final String ideviceIdPath = path.join(getArtifactPath(), 'libimobiledevice', 'idevice_id');
-    final List<String> iosDeviceIDs = LineSplitter.split(await eval(ideviceIdPath, <String>['-l'], environment: _ideviceIdEnvironment))
-      .map<String>((String line) => line.trim())
-      .where((String line) => line.isNotEmpty)
-      .toList();
-    if (iosDeviceIDs.isEmpty)
+    final List<dynamic> results = json.decode(await eval(
+      path.join(flutterDirectory.path, 'bin', 'flutter'),
+      <String>['devices', '--machine', '--suppress-analytics'],
```

Comment:
I would also pass --no-version-check just to be safe
https://github.com/flutter/flutter/blob/75fd73d81888966746821e65d37e1fc90dc56bfb/packages/flutter_tools/lib/src/runner/flutter_command_runner.dart#L82

Reply:
Sorry, landed this while the tree was green. Fortunately, in the command runner we bypass the version check if `--machine` is provided https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/runner/flutter_command_runner.dart#L82